src/core.rs

```diff
@@ -431,9 +434,370 @@ impl Projectm {
         // a pointer to it.
         let ptr = path
             .as_ref()
-            .map(|s| s.as_ptr() as *const i8)
+            .map(|s| s.as_ptr())
             .unwrap_or(std::ptr::null());
 
         unsafe { ffi::projectm_write_debug_image_on_next_frame(instance, ptr) };
     }
 }
+
+pub struct ProjectM {
+    instance: Rc<RefCell<ProjectMHandle>>,
```

Comment:
is it Rc instead of Arc because a projectM instance can't be shared between threads?
i am wondering how to best handle audio capture callbacks when the main thread in projectm-sdl has the `ProjectM` instance but the audio callback needs to feed it data. should the callback send the data to the main thread or can the instance be shared?
(I don't know what I'm doing at all)